rosidl_generator_c/rosidl_generator_c/__init__.py

```diff
@@ -209,9 +222,13 @@ def basic_value_to_c(type_, value):
         return f'{value}ull'
 
     if 'float' == type_.typename:
+        if isnan(float(value)):
+            return 'NAN'
```

Comment:
I chose to use `'nanf(\"\")'` on my PR to avoid the c-style cast to a double later.